src/collar/oc_states.lsl

```diff
@@ -187,35 +184,35 @@ default
         g_iExpectAlive=1;
         llSetTimerEvent(1);
         //llScriptProfiler(TRUE);
-        llMessageLinked(LINK_SET, REBOOT,"reboot", "");
-        llSleep(5);
+      //  llMessageLinked(LINK_SET, REBOOT,"reboot", "");
+        //llSleep(5);
         //llMessageLinked(LINK_SET, 0, "initialize", "");
         if(g_iVerbosityLevel>=1)
             llOwnerSay("Collar is preparing to startup, please be patient.");
     }
     
     
     on_rez(integer iRez){
-        llSleep(10);
+        //llSleep(10);
         llResetScript();
     }
     
     timer(){
         
         if(g_iExpectAlive){
-            if(llGetTime()>=5 && g_iPasses<3){
+            if(llGetTime()>=3 && g_iPasses<2){ //>=5
                 llMessageLinked(LINK_SET,READY, "","");
                 llResetTime();
                 //llSay(0, "PASS COUNT: "+(string)g_iPasses);
                 g_iPasses++;
-            } else if(llGetTime()>=7.5 && g_iPasses>=3){
+            } else if(llGetTime()>=2 && g_iPasses>=2){ //>=7.5
                 if(g_iVerbosityLevel>=2)
                     llOwnerSay("Scripts ready: "+(string)llGetListLength(g_lAlive));
                 llMessageLinked(LINK_SET,STARTUP,llDumpList2String(g_lAlive,","),"");
                 g_iExpectAlive=0;
                 g_lAlive=[];
                 g_iPasses=0;
-                llSleep(10);
+                llSleep(1); //10
```

Comment:
I know you didn't write it, but I _hate_ code with arbitrary sleeps to let other things finish.  It puts us in this horrible position of having to choose between waiting forever or risking that one of those things isn't done.

What are we waiting for here?  Is there a way we can get notified of that thing or check for it instead of just sleeping?

Reply:
Essentially this is looping through a process where the scripts report back an alive status and leaving a cushion to ensure that every script has had a chance to do its own startup things before settings start getting sent out.

There's no straightforward way to receive notifications of this. As of now we're basically setting things running and moving ahead when there haven't been any new scripts reporting that they have come alive for a little while. Some kind of padding here is necessary unless we implement a method for every script to report not just an alive state but a ready state. This method isn't particularly scientific, but this tuning of the timing has worked so far in testing. 

We could beneficially redo the entire startup process, but that's a pretty major project.